[GEOT-5302] remove support for EnvFunction lookup tables

fgdrf opened the pull request:
see https://osgeo-org.atlassian.net/browse/GEOT-5302 for details and comments.

Implemented option 1 and added two functions that provides remove support. In addition I updated the javadoc comment for setGlobalValue

jodygarnett commented on modules/library/main/src/main/java/org/geotools/filter/function/EnvFunction.java:
```diff
@@ -90,6 +90,7 @@
  *
```
update header

jodygarnett commented:
Do we have any documentation on how EnvFunctions work that should be updated as part of this pull request?

fgdrf commented:
Update javadoc and added serveral SLD & code java examples how to use isNull and isNil. Added serveral test cases to verify behavior.
EnvFunction.setGlobalValue() now supports null values, changed internal Map from ConcurrentHashMap to a synchronized HashMap to be thread save for gets and puts.

jodygarnett commented on modules/library/main/src/main/java/org/geotools/filter/function/EnvFunction.java:
```diff
@@ -20,12 +20,12 @@
 import static org.geotools.filter.capability.FunctionNameImpl.parameter;
```
Please update header

jodygarnett commented:
Other then updating headers this one is good to go.

bencaradocdavies commented:
@fgdrf I think this was manually merged by @jodygarnett on 14.x and master.
